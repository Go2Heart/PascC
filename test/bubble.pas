program BubbleSort;

var
  arr: array[1..10] of integer;
  i, j, temp: integer;

begin
  writeln('Enter 10 numbers:');
  for i := 1 to 10 do
    readln(arr[i]);

  for i := 1 to 9 do
    for j := i + 1 to 10 do
      if arr[i] > arr[j] then begin
        temp := arr[i];
        arr[i] := arr[j];
        arr[j] := temp;
      end;

  writeln('Sorted array:');
  for i := 1 to 10 do
    write(arr[i], ' ');
end.