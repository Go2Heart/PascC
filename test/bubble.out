LexToken(PROGRAM,'program',1,0)
LexToken(ID,'BubbleSort',1,8)
LexToken(SEMI,';',1,18)
LexToken(VAR,'var',3,21)
LexToken(ID,'arr',4,27)
LexToken(COLON,':',4,30)
LexToken(ARRAY,'array',4,32)
LexToken(LBRACK,'[',4,37)
LexToken(ICONST,1,4,38)
LexToken(DOTDOT,'..',4,39)
LexToken(ICONST,10,4,41)
LexToken(RBRACK,']',4,43)
LexToken(OF,'of',4,45)
LexToken(INTEGER,'integer',4,48)
LexToken(SEMI,';',4,55)
LexToken(ID,'i',5,59)
LexToken(COMMA,',',5,60)
LexToken(ID,'j',5,62)
LexToken(COMMA,',',5,63)
LexToken(ID,'temp',5,65)
LexToken(COLON,':',5,69)
LexToken(INTEGER,'integer',5,71)
LexToken(SEMI,';',5,78)
LexToken(BEGIN,'begin',7,81)
LexToken(ID,'writeln',8,89)
LexToken(LPAREN,'(',8,96)
LexToken(STRING,'Enter 10 numbers:',8,97)
LexToken(RPAREN,')',8,116)
LexToken(SEMI,';',8,117)
LexToken(FOR,'for',9,121)
LexToken(ID,'i',9,125)
LexToken(ASSIGN,':=',9,127)
LexToken(ICONST,1,9,130)
LexToken(TO,'to',9,132)
LexToken(ICONST,10,9,135)
LexToken(DO,'do',9,138)
LexToken(ID,'readln',10,145)
LexToken(LPAREN,'(',10,151)
LexToken(ID,'arr',10,152)
LexToken(LBRACK,'[',10,155)
LexToken(ID,'i',10,156)
LexToken(RBRACK,']',10,157)
LexToken(RPAREN,')',10,158)
LexToken(SEMI,';',10,159)
LexToken(FOR,'for',12,164)
LexToken(ID,'i',12,168)
LexToken(ASSIGN,':=',12,170)
LexToken(ICONST,1,12,173)
LexToken(TO,'to',12,175)
LexToken(ICONST,9,12,178)
LexToken(DO,'do',12,180)
LexToken(FOR,'for',13,187)
LexToken(ID,'j',13,191)
LexToken(ASSIGN,':=',13,193)
LexToken(ID,'i',13,196)
LexToken(ADDOP,'+',13,198)
LexToken(ICONST,1,13,200)
LexToken(TO,'to',13,202)
LexToken(ICONST,10,13,205)
LexToken(DO,'do',13,208)
LexToken(IF,'if',14,217)
LexToken(ID,'arr',14,220)
LexToken(LBRACK,'[',14,223)
LexToken(ID,'i',14,224)
LexToken(RBRACK,']',14,225)
LexToken(RELOP,'>',14,227)
LexToken(ID,'arr',14,229)
LexToken(LBRACK,'[',14,232)
LexToken(ID,'j',14,233)
LexToken(RBRACK,']',14,234)
LexToken(THEN,'then',14,236)
LexToken(BEGIN,'begin',14,241)
LexToken(ID,'temp',15,255)
LexToken(ASSIGN,':=',15,260)
LexToken(ID,'arr',15,263)
LexToken(LBRACK,'[',15,266)
LexToken(ID,'i',15,267)
LexToken(RBRACK,']',15,268)
LexToken(SEMI,';',15,269)
LexToken(ID,'arr',16,279)
LexToken(LBRACK,'[',16,282)
LexToken(ID,'i',16,283)
LexToken(RBRACK,']',16,284)
LexToken(ASSIGN,':=',16,286)
LexToken(ID,'arr',16,289)
LexToken(LBRACK,'[',16,292)
LexToken(ID,'j',16,293)
LexToken(RBRACK,']',16,294)
LexToken(SEMI,';',16,295)
LexToken(ID,'arr',17,305)
LexToken(LBRACK,'[',17,308)
LexToken(ID,'j',17,309)
LexToken(RBRACK,']',17,310)
LexToken(ASSIGN,':=',17,312)
LexToken(ID,'temp',17,315)
LexToken(SEMI,';',17,319)
LexToken(END,'end',18,327)
LexToken(SEMI,';',18,330)
LexToken(ID,'writeln',20,335)
LexToken(LPAREN,'(',20,342)
LexToken(STRING,'Sorted array:',20,343)
LexToken(RPAREN,')',20,358)
LexToken(SEMI,';',20,359)
LexToken(FOR,'for',21,363)
LexToken(ID,'i',21,367)
LexToken(ASSIGN,':=',21,369)
LexToken(ICONST,1,21,372)
LexToken(TO,'to',21,374)
LexToken(ICONST,10,21,377)
LexToken(DO,'do',21,380)
LexToken(WRITE,'write',22,387)
LexToken(LPAREN,'(',22,392)
LexToken(ID,'arr',22,393)
LexToken(LBRACK,'[',22,396)
LexToken(ID,'i',22,397)
LexToken(RBRACK,']',22,398)
LexToken(COMMA,',',22,399)
LexToken(CCONST,"' '",22,401)
LexToken(RPAREN,')',22,404)
LexToken(SEMI,';',22,405)
LexToken(END,'end',23,407)
LexToken(DOT,'.',23,410)
